spec/lib/api/filter_spec.rb

```diff
@@ -406,4 +406,52 @@
       expect(actual.exp).to eq(expected)
     end
   end
+
+  describe "#split_filter_string (private)" do
+    let(:model) { Vm }
+
+    {
+      # all operators, and a few curve balls
+      "name=1"      => [[], "name", "=", "1"],
+      "name=>1"     => [[], "name", "=", ">1"],
+      "name<1"      => [[], "name", "<", "1"],
+      "name<<1"     => [[], "name", "<", "<1"],
+      "name>1"      => [[], "name", ">", "1"],
+      "name>>1"     => [[], "name", ">", ">1"],
+      "name==1"     => [[], "name", "==", "1"],
+      "name!=1"     => [[], "name", "!=", "1"],
+      "name=~1"     => [[], "name", "=~", "1"],
+      "name!~1"     => [[], "name", "!~", "1"],
+      "name!==1"    => [[], "name", "!==", "1"],
+      # funky formatting
+      " name = x "  => [[], "name", "=", "x"],
+      "name=  "     => [[], "name", "=", ""],
+      "name= ' ' "  => [[], "name", "=", "' '"],
+      "host.name=x" => [["host"], "name", "=", "x"],
+    }.each do |str, (associations, column, eoperator, evalue)|
+      it "parses #{str}" do
+        field, operator, value = split_filter_string(str)
+
+        # currently, can not compare one MiqExpression::Target with another
```

Comment:
```suggestion
        # currently, cannot compare one MiqExpression::Target with another
```

On that note, does it make sense to implement `==` on MiqExpression::Target or is that incorrect?

Reply:
yea. I had forgotten to link it